✏️ added docs for benchmarking (#188)
kids-first/kf-somatic-workflow: docs/SOMATIC_SNV_BENCHMARK.md

1. Gold standard VCFs were downloaded from https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/seqc/Somatic_Mutation_WG/release/v1.2/ with [SNV](https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/seqc/Somatic_Mutation_WG/release/v1.2/high-confidence_sSNV_in_HC_regions_v1.2.vcf.gz) and [INDEL](https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/seqc/Somatic_Mutation_WG/release/v1.2/high-confidence_sINDEL_in_HC_regions_v1.2.vcf.gz) VCFs merged into a single VCF
1. Merged gold standard VCF was then subset to BWA hits and broken up into call sets matching samples (parse `calledSamples` `INFO` field)

> 1. Next, a custom [benchmark script](https://github.com/kids-first/kfdrc-benchmark/blob/main/scripts/benchmark_vcf_calls.py), which collected the following metrics:
>  - True Positive
>  - False Positive
>  - False Negative
>  - F1 Score
>  - Precision
>  - Accuracy

Reviewer: I think you can make these sub-bullets by indenting more
```suggestion
1. Next, a custom [benchmark script](https://github.com/kids-first/kfdrc-benchmark/blob/main/scripts/benchmark_vcf_calls.py), which collected the following metrics:
   - True Positive
   - False Positive
   - False Negative
   - F1 Score
   - Precision
   - Accuracy
```

Author: I had a feeling you'd point that out. They look different which is a little annoying, but probably better than having the same indent

To summarize their work:
Their starting data consists of 21 Replicates distributed among 5 sequencing centers aligned with 3 different aligners (BWA, Bowtie2, NovoAlign). All the aligned BAMs were called with 6 callers (Mutect2, SomaticSniper, VarDict, MuSE, Strelka2, TNscope) resulting into 63 VCFs in total. All the calls were merged together to create a consensus VCF. This consensus file was fed into a deep learning classifier (SomaticSeq, NeuSomatic) that was trained against truth VCFs. These truth VCFs were produced by spiking mutations randomly in the reads. This ML model classify these calls into 4 confidence levels, in which we report the average number of PASS calls given by the classifier in each VCF, as well as REJECT:
 - High Confidence: 60.5 PASS, 0.15 REJECT and high VAF.
 - Medium Confidence: 27.4 PASS, 0.80 REJECT and low VAF not captured in many replicates.
 - Low Confidence: 8.2 PASS, 1.7 REJECT and very low VAF, cannot distinguish from noise.
 - Unclassified: 1.8 PASS and 3.8 REJECT and likely false positive

> A helpful [YouTube video](https://www.youtube.com/watch?v=pDsEo0xdHWA&t=1s) also exists to explain their methodology.

Reviewer: ```suggestion
A helpful [YouTube video](https://www.youtube.com/watch?v=pDsEo0xdHWA) also exists to explain their methodology.
```

## Comparison to "Gold Standard" Dataset

> BAM files relevant to our workflow (BWA-aligned) were called using our standard [soamtic workflow](https://github.com/kids-first/kf-somatic-workflow/releases/tag/v4.4.2) followed up by our [consensus caller](https://github.com/kids-first/kf-somatic-workflow/blob/v4.3.5/workflow/kfdrc_consensus_calling.cwl). Synthetic BAM files did need to have read groups corrected as `SM` identifier was left to match the normal sample that was used for spike-in instead of modifying it for a proper tumor-normal comparison. Gold standard VCFs provided by the authors all all samples aggregated. To make 1:1 comparisons, we did the following:

Reviewer: ```suggestion
BAM files relevant to our workflow (BWA-aligned) were called using our standard [somatic workflow](https://github.com/kids-first/kf-somatic-workflow/releases/tag/v4.4.2) followed up by our [consensus caller](https://github.com/kids-first/kf-somatic-workflow/blob/v4.3.5/workflow/kfdrc_consensus_calling.cwl). Synthetic BAM files did need to have read groups corrected. Originally, their `SM` identifier matched the normal sample used as the starting point for the spike-in experiment. We updated the `SM` field to match the synthetic BAM filename (e.g. `syn_<normal_sample_name>`). The gold standard VCFs provided by the authors had all samples aggregated. To make 1:1 comparisons, we did the following:
```

Reviewer: Why are we linking to the release for `somatic_workflow`? Should we link to the specific workflow in that release that we used ala: https://github.com/kids-first/kf-somatic-workflow/blob/v4.4.2/workflow/kfdrc-somatic-variant-workflow.cwl

Author: Hmmm, I thought I had. That was an oversight

> 1. Gold standard VCFs were downloaded from https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/seqc/Somatic_Mutation_WG/release/v1.2/ with [SNV](https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/seqc/Somatic_Mutation_WG/release/v1.2/high-confidence_sSNV_in_HC_regions_v1.2.vcf.gz) and INDEL VCFs merged into a single VCF

Reviewer: ```suggestion
1. Gold standard VCFs were downloaded from https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/seqc/Somatic_Mutation_WG/release/v1.2/ with [SNV](https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/seqc/Somatic_Mutation_WG/release/v1.2/high-confidence_sSNV_in_HC_regions_v1.2.vcf.gz) and [INDEL](https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/seqc/Somatic_Mutation_WG/release/v1.2/high-confidence_sINDEL_in_HC_regions_v1.2.vcf.gz) VCFs merged into a single VCF
```

Reviewer: Also, did we use the high confidence VCFs or did we just use the raw superset for these comparisons?

Author: I am going to say High Conf VCFs because those are the links I had here: https://www.notion.so/d3b/SNV-Benchmarking-Project-352f4bb803d04498858cf1a6403c33c8?pvs=4#bb64dfe23cbe4b9d9023bda0a21c9e87 and I think when I wrote this doc, I was much closer to the work being done

Reviewer: Yeah but looking at the comparisons, we have `vs superset (high + med + low + unclassified)`. I don't think there are med, low, and unclassified variants in the high confidence VCF.

Author: Ah, good point. Perhaps started off with high conf, but then bounced around in order to do some extra analyses. High conf file does have the medium in it:
```
PASS;HighConf
PASS;MedConf
```
but yeah, the low ones are not. I can update the link

2. Merged gold standard VCF was then subset to BWA hits and broken up into call sets matching samples (parse `calledSamples` `INFO` field)

> Next, a custom [benchmark script](https://github.com/kids-first/kfdrc-benchmark/blob/main/scripts/benchmark_vcf_calls.py), which collected the following metrics:

Reviewer: Should this be in the numbered list?
